src/core/Dropdown/Dropdown/Dropdown.tsx

```diff
@@ -1,49 +1,79 @@
-import React, {
-  Component,
-  ReactNode,
-  ReactElement,
-  forwardRef,
-  useRef,
-} from 'react';
+import React, { Component, ReactNode, ReactElement, forwardRef } from 'react';
 import { default as styled } from 'styled-components';
 import classnames from 'classnames';
-import {
-  ListboxInput,
-  ListboxButton,
-  ListboxButtonProps,
-  ListboxPopoverProps,
-  ListboxList,
-  ListboxPopover,
-  useListboxContext,
-} from '@reach/listbox';
-import { positionMatchWidth } from '@reach/popover';
 import { getConditionalAriaProp } from '../../../utils/aria';
-import { useEnhancedEffect } from '../../../utils/common';
 import { getLogger } from '../../../utils/log';
 import { AutoId } from '../../utils/AutoId/AutoId';
-import { HtmlSpan, HtmlDiv } from '../../../reset';
+import { HtmlSpan, HtmlDiv, HtmlInput } from '../../../reset';
 import { Label, LabelMode } from '../../Form/Label/Label';
 import { DropdownItemProps } from '../DropdownItem/DropdownItem';
 import { baseStyles } from './Dropdown.baseStyles';
 import { SuomifiThemeProp, SuomifiThemeConsumer } from '../../theme';
+import {
+  forkRefs,
+  getOwnerDocument,
+  getRecursiveChildText,
+} from '../../../utils/common/common';
+import { Popover } from '../../../core/Popover/Popover';
+import { SelectItemList } from '../../Form/Select/BaseSelect/SelectItemList/SelectItemList';
+import { HintText } from '../../Form/HintText/HintText';
+import { StatusText } from '../../Form/StatusText/StatusText';
+import { StatusTextCommonProps } from '../../Form/types';
+
+// Dropdown is a button but it also has a listbox popup
+/* eslint-disable jsx-a11y/role-supports-aria-props */
 
 const baseClassName = 'fi-dropdown';
 
 export const dropdownClassNames = {
   baseClassName,
   labelIsVisible: `${baseClassName}_label--visible`,
+  inputWrapper: `${baseClassName}_input-wrapper`,
   button: `${baseClassName}_button`,
   popover: `${baseClassName}_popover`,
+  itemList: `${baseClassName}_item-list`,
   item: `${baseClassName}_item`,
-  noSelectedStyles: `${baseClassName}--noSelectedStyles`,
+  statusTextHasContent: `${baseClassName}_statusText--has-content`,
+  open: `${baseClassName}--open`,
   disabled: `${baseClassName}--disabled`,
+  error: `${baseClassName}--error`,
+  italicize: `${baseClassName}--italicize`,
+};
+
+export interface DropdownProviderState {
+  /** Callback for communicating DropdownItem click to parent  */
+  onItemClick: (itemValue: string) => void;
+  /** Currently selected DropdownItem */
+  selectedDropdownValue: string | undefined | null;
+  /** Currently focused DropdownItem */
+  focusedItemID: string | null | undefined;
```

Comment:
I think this should be focusedItemId, so props are in camelCase

Reply:
Fixd